README.rst

```diff
@@ -1965,6 +1965,77 @@ The ``AlertSettings`` of ``ping`` metric will by default use ``threshold`` and `
 defined in the ``alert_settings`` key.
 You can always override them and define your own custom values via the *admin*.
 
+You can also use the ``alert_field`` key in metric configuration
+which allows ``AlertSettings`` to check the ``threshold`` on
+``alert_field`` instead of the default ``field_name`` key.
+
+In the following example, we added ``alert_field`` to the metric configuration
+for the iperf metric in order to check the threshold on **iperf bandwidth**.
+
+.. code-block:: python
+
+    DEFAULT_METRICS = {
+        'iperf': {
+            'label': _('Iperf'),
+            'name': 'Iperf',
+            'key': 'iperf',
+            'field_name': 'iperf_result',
+            'related_fields': [
+                'sent_bps_tcp',
+                'received_bps_tcp',
+                'sent_bytes_tcp',
+                'received_bytes_tcp',
+                'retransmits',
+                'sent_bytes_udp',
+                'sent_bps_udp',
+                'jitter',
+                'total_packets',
+                'lost_packets',
+                'lost_percent',
+            ],
+            # Check alert threshold on iperf bandwidth (TCP)
+            'alert_field': 'sent_bps_tcp',
+            # Alert is sent immediately (tolerance 0) when bandwidth less than 10 Mbps
+            'alert_settings': {'operator': '<', 'threshold': 10000000, 'tolerance': 0},
+            'notification': {
+                'problem': {
+                    'verbose_name': 'Iperf PROBLEM',
+                    'verb': _('Iperf bandwidth is less than normal value'),
+                    'level': 'warning',
+                    'email_subject': _(
+                        '[{site.name}] PROBLEM: {notification.target} {notification.verb}'
+                    ),
+                    'message': _(
+                        'The device [{notification.target}]({notification.target_link}) '
+                        '{notification.verb}.'
+                    ),
+                },
+                'recovery': {
+                    'verbose_name': 'Iperf RECOVERY',
+                    'verb': _('Iperf bandwidth now back to normal'),
+                    'level': 'info',
+                    'email_subject': _(
+                        '[{site.name}] RECOVERY: {notification.target} {notification.verb}'
+                    ),
+                    'message': _(
+                        'The device [{notification.target}]({notification.target_link}) '
+                        '{notification.verb}.'
+                    ),
+                },
+            },
+            # iperf charts configuration here
+        },
+    }
```

Comment:
is it really necessary to redefine all this?

Wouldn't the following be enough?

```python
# Main project settings.py

DEFAULT_METRICS = {
    'iperf': {
        'alert_field': 'sent_bps_tcp',
        # Alert is sent immediately (tolerance 0) when bandwidth less than 10 Mbps
        'alert_settings': {'operator': '<', 'threshold': 10000000, 'tolerance': 2880},
    },
    'notification': {
        'problem': {
            'verbose_name': 'Bandwidth PROBLEM',
            'verb': _('Iperf bandwidth is less than normal value'),
        },
        'recovery': {
            'verbose_name': 'Bandwidth RECOVERY',
            'verb': _('Iperf bandwidth now back to normal'),
        },
    },
}
```

Please double check this and test the pseudo code above.

Reply:
We need to create `AlertSettings` for `iperf check` first, then we can add the above configuration to `settings.py`.